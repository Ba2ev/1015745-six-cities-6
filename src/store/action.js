export const ActionType = {
  UPDATE_CITY: `city/updateCity`,
  UPDATE_SORT: `sort/updateSort`,
  TOGGLE_SORTLIST: `sort/toggleSortList`,
  UPDATE_HOVERED_OFFER_ID: `offer/updateHoveredOfferId`,
  LOAD_OFFERS: `offers/loadOffers`,
  LOAD_FAVORITES_OFFERS: `offers/loadFavoritesOffers`,
  UPDATE_FAVORITES_OFFERS: `offers/updateFavoritesOffers`,
  LOAD_OFFER_DATA: `offer/loadOfferData`,
  LOAD_OFFER_COMMENTS: `offer/loadOfferComments`,
  LOAD_OFFER_NEARBY: `offer/loadOfferNearby`,
  REQUIRED_AUTHORIZATION: `user/requiredAuthorization`,
  UPDATE_ACCOUNT: `user/updateAccount`,
};

export const ActionCreator = {
  updateCity: (city) => ({
    type: ActionType.UPDATE_CITY,
    payload: city
  }),
  updateSort: (sortType) => ({
    type: ActionType.UPDATE_SORT,
    payload: sortType
  }),
  toggleSortList: () => ({
    type: ActionType.TOGGLE_SORTLIST,
  }),
  updateHoveredOfferId: (id) => ({
    type: ActionType.UPDATE_HOVERED_OFFER_ID,
    payload: id
  }),
  loadOffers: (offers) => ({
    type: ActionType.LOAD_OFFERS,
    payload: offers
  }),
  loadFavoritesOffer: (offers) => ({
    type: ActionType.LOAD_FAVORITES_OFFERS,
    payload: offers
  }),
  loadOfferNearby: (nearblyOffers) => ({
    type: ActionType.LOAD_OFFER_NEARBY,
    payload: nearblyOffers
  }),
  loadOfferData: (offerData) => ({
    type: ActionType.LOAD_OFFER_DATA,
    payload: offerData
  }),
  loadOfferComments: (offerComments) => ({
    type: ActionType.LOAD_OFFER_COMMENTS,
    payload: offerComments
  }),
  updateFavoriteOffers: (offerData) => ({
    type: ActionType.UPDATE_FAVORITES_OFFERS,
    payload: offerData
  }),
  requireAuthorization: (status) => ({
    type: ActionType.REQUIRED_AUTHORIZATION,
    payload: status,
  }),
  updateAccount: (email) => ({
    type: ActionType.UPDATE_ACCOUNT,
    payload: email,
  }),
};

