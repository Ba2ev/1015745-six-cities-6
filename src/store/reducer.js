import {combineReducers} from 'redux';
import {user} from './user/user';
import {property} from './property/property';
import {places} from './places/places';

export const NameSpace = {
  PROPERTY: `PROPERTY`,
  PLACES: `PLACES`,
  USER: `USER`
};

export default combineReducers({
  [NameSpace.PROPERTY]: property,
  [NameSpace.PLACES]: places,
  [NameSpace.USER]: user,
});
