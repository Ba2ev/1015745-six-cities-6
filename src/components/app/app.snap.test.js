import React from "react";
import {render} from '@testing-library/react';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import App from './app';
import {NameSpace} from '../../store/reducer';

const mockStore = configureStore({});

describe(`Test routing`, () => {
  jest.spyOn(redux, `useSelector`);
  jest.spyOn(redux, `useDispatch`);

  it(`Render MainPage when user navigate to '/' url`, () => {
    const history = createMemoryHistory();

    const offers = [
      {
        "city": {
          "name": `Paris`,
          "location": {
            "latitude": 48.85661,
            "longitude": 2.351499,
            "zoom": 13
          }
        },
        "previewImage": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
        "images": [
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`
        ],
        "title": `The house among olive `,
        "isFavorite": true,
        "isPremium": true,
        "rating": 3,
        "type": `room`,
        "bedrooms": 1,
        "maxAdults": 1,
        "price": 169,
        "goods": [
          `Breakfast`,
          `Air conditioning`,
          `Laptop friendly workspace`,
          `Washer`
        ],
        "host": {
          "id": 25,
          "name": `Angelina`,
          "isPro": true,
          "avatarUrl": `img/avatar-angelina.jpg`
        },
        "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
        "location": {
          "latitude": 48.83861,
          "longitude": 2.350499,
          "zoom": 16
        },
        "id": 1
      },
      {
        "city": {
          "name": `Brussels`,
          "location": {
            "latitude": 50.846557,
            "longitude": 4.351697,
            "zoom": 13
          }
        },
        "previewImage": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
        "images": [
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`
        ],
        "title": `Waterfront with extraordinary view`,
        "isFavorite": false,
        "isPremium": false,
        "rating": 4.8,
        "type": `apartment`,
        "bedrooms": 5,
        "maxAdults": 6,
        "price": 310,
        "goods": [
          `Dishwasher`,
          `Towels`,
          `Washer`,
          `Breakfast`,
          `Baby seat`,
          `Air conditioning`,
          `Fridge`,
          `Laptop friendly workspace`,
          `Coffee machine`
        ],
        "host": {
          "id": 25,
          "name": `Angelina`,
          "isPro": true,
          "avatarUrl": `img/avatar-angelina.jpg`
        },
        "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
        "location": {
          "latitude": 50.844556999999995,
          "longitude": 4.346697,
          "zoom": 16
        },
        "id": 2
      },
      {
        "city": {
          "name": `Cologne`,
          "location": {
            "latitude": 50.938361,
            "longitude": 6.959974,
            "zoom": 13
          }
        },
        "previewImage": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
        "images": [
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`
        ],
        "title": `Penthouse, 4-5 rooms + 5 balconies`,
        "isFavorite": true,
        "isPremium": true,
        "rating": 4.2,
        "type": `hotel`,
        "bedrooms": 4,
        "maxAdults": 6,
        "price": 182,
        "goods": [
          `Laptop friendly workspace`,
          `Washer`,
          `Breakfast`,
          `Air conditioning`
        ],
        "host": {
          "id": 25,
          "name": `Angelina`,
          "isPro": true,
          "avatarUrl": `img/avatar-angelina.jpg`
        },
        "description": `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
        "location": {
          "latitude": 50.950361,
          "longitude": 6.961974,
          "zoom": 16
        },
        "id": 3
      },
    ];

    const initialState = {
      [NameSpace.USER]: {
        isAuthorized: false,
      },
      [NameSpace.DATA]: {
        city: `Paris`,
        offers,
        isOffersLoaded: true,
      }
    };

    const {container} = render(
        <redux.Provider store={mockStore(initialState)}>
          <Router history={history}>
            <App />
          </Router>
        </redux.Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`Render FavoritesPage when user navigate to '/favorites' url`, () => {
    const history = createMemoryHistory();

    const favorites = [
      {
        "city": {
          "name": `Paris`,
          "location": {
            "latitude": 48.85661,
            "longitude": 2.351499,
            "zoom": 13
          }
        },
        "previewImage": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
        "images": [
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`
        ],
        "title": `The house among olive `,
        "isFavorite": true,
        "isPremium": true,
        "rating": 3,
        "type": `room`,
        "bedrooms": 1,
        "maxAdults": 1,
        "price": 169,
        "goods": [
          `Breakfast`,
          `Air conditioning`,
          `Laptop friendly workspace`,
          `Washer`
        ],
        "host": {
          "id": 25,
          "name": `Angelina`,
          "isPro": true,
          "avatarUrl": `img/avatar-angelina.jpg`
        },
        "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
        "location": {
          "latitude": 48.83861,
          "longitude": 2.350499,
          "zoom": 16
        },
        "id": 1
      },
      {
        "city": {
          "name": `Cologne`,
          "location": {
            "latitude": 50.938361,
            "longitude": 6.959974,
            "zoom": 13
          }
        },
        "previewImage": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
        "images": [
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
          `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`
        ],
        "title": `Penthouse, 4-5 rooms + 5 balconies`,
        "isFavorite": true,
        "isPremium": true,
        "rating": 4.2,
        "type": `hotel`,
        "bedrooms": 4,
        "maxAdults": 6,
        "price": 182,
        "goods": [
          `Laptop friendly workspace`,
          `Washer`,
          `Breakfast`,
          `Air conditioning`
        ],
        "host": {
          "id": 25,
          "name": `Angelina`,
          "isPro": true,
          "avatarUrl": `img/avatar-angelina.jpg`
        },
        "description": `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
        "location": {
          "latitude": 50.950361,
          "longitude": 6.961974,
          "zoom": 16
        },
        "id": 3
      },
    ];

    const initialState = {
      [NameSpace.USER]: {
        isAuthorized: true,
        account: `test@test.ru`,
      },
      [NameSpace.DATA]: {
        favorites,
        isFavoritesLoaded: true,
      }
    };

    history.push(`/favorites`);

    const {container} = render(
        <redux.Provider store={mockStore(initialState)}>
          <Router history={history}>
            <App />
          </Router>
        </redux.Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`Render LoginPage when user navigate to '/login' url`, () => {
    const history = createMemoryHistory();


    const initialState = {
      [NameSpace.USER]: {
        isAuthorized: false,
      }
    };

    history.push(`/login`);

    const {container} = render(
        <redux.Provider store={mockStore(initialState)}>
          <Router history={history}>
            <App />
          </Router>
        </redux.Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`Render 'NotFoundPage' when user navigate to non-existent route`, () => {
    const history = createMemoryHistory();

    const initialState = {
      [NameSpace.USER]: {
        isAuthorized: true,
        account: `test@test.ru`,
      }
    };

    history.push(`/non-existent-route`);

    const {container} = render(
        <redux.Provider store={mockStore(initialState)}>
          <Router history={history}>
            <App />
          </Router>
        </redux.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
