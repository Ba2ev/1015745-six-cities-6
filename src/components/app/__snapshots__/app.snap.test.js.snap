// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test routing Render 'NotFoundPage' when user navigate to non-existent route 1`] = `
<div>
  <h1>
    404. Page not found
  </h1>
  <a
    href="/"
  >
    Вернуться на главную
  </a>
</div>
`;

exports[`Test routing Render FavoritesPage when user navigate to '/favorites' url 1`] = `
<div>
  <div
    class="page "
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="81"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__user-name user__name"
                  >
                    test@test.ru
                  </span>
                </a>
                <a>
                  Sign out
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--favorites "
    >
      <div
        class="page__favorites-container container"
      >
        <section
          class="favorites"
        >
          <h1
            class="favorites__title"
          >
            Saved listing
          </h1>
          <ul
            class="favorites__list"
          >
            <li
              class="favorites__locations-items"
            >
              <div
                class="favorites__locations locations locations--current"
              >
                <div
                  class="locations__item"
                >
                  <a
                    class="locations__item-link"
                    href="#"
                  >
                    <span>
                      Cologne
                    </span>
                  </a>
                </div>
              </div>
              <div
                class="favorites__places"
              >
                <article
                  class="favorites__card place-card"
                  data-cart-id="3"
                >
                  <div
                    class="place-card__mark"
                  >
                    <span>
                      Premium
                    </span>
                  </div>
                  <div
                    class="favorites__image-wrapper place-card__image-wrapper"
                  >
                    <a
                      href="/offer/3"
                    >
                      <img
                        alt="Place image"
                        class="place-card__image"
                        height="110"
                        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg"
                        width="150"
                      />
                    </a>
                  </div>
                  <div
                    class="favorites__card-info place-card__info"
                  >
                    <div
                      class="place-card__price-wrapper"
                    >
                      <div
                        class="place-card__price"
                      >
                        <b
                          class="place-card__price-value"
                        >
                          €
                          182
                        </b>
                        <span
                          class="place-card__price-text"
                        >
                          / night
                        </span>
                      </div>
                      <button
                        class="place-card__bookmark-button place-card__bookmark-button--active button"
                        type="button"
                      >
                        <svg
                          class="place-card__bookmark-icon"
                          height="19"
                          width="18"
                        >
                          <use
                            xlink:href="#icon-bookmark"
                          />
                        </svg>
                        <span
                          class="visually-hidden"
                        >
                          In bookmarks
                        </span>
                      </button>
                    </div>
                    <div
                      class="place-card__rating rating"
                    >
                      <div
                        class="place-card__stars rating__stars"
                      >
                        <span
                          style="width: 84%;"
                        />
                        <span
                          class="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <h2
                      class="place-card__name"
                    >
                      <a
                        href="/offer/3"
                      >
                        Penthouse, 4-5 rooms + 5 balconies
                      </a>
                    </h2>
                    <p
                      class="place-card__type"
                    >
                      hotel
                    </p>
                  </div>
                </article>
              </div>
            </li>
            <li
              class="favorites__locations-items"
            >
              <div
                class="favorites__locations locations locations--current"
              >
                <div
                  class="locations__item"
                >
                  <a
                    class="locations__item-link"
                    href="#"
                  >
                    <span>
                      Paris
                    </span>
                  </a>
                </div>
              </div>
              <div
                class="favorites__places"
              >
                <article
                  class="favorites__card place-card"
                  data-cart-id="1"
                >
                  <div
                    class="place-card__mark"
                  >
                    <span>
                      Premium
                    </span>
                  </div>
                  <div
                    class="favorites__image-wrapper place-card__image-wrapper"
                  >
                    <a
                      href="/offer/1"
                    >
                      <img
                        alt="Place image"
                        class="place-card__image"
                        height="110"
                        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
                        width="150"
                      />
                    </a>
                  </div>
                  <div
                    class="favorites__card-info place-card__info"
                  >
                    <div
                      class="place-card__price-wrapper"
                    >
                      <div
                        class="place-card__price"
                      >
                        <b
                          class="place-card__price-value"
                        >
                          €
                          169
                        </b>
                        <span
                          class="place-card__price-text"
                        >
                          / night
                        </span>
                      </div>
                      <button
                        class="place-card__bookmark-button place-card__bookmark-button--active button"
                        type="button"
                      >
                        <svg
                          class="place-card__bookmark-icon"
                          height="19"
                          width="18"
                        >
                          <use
                            xlink:href="#icon-bookmark"
                          />
                        </svg>
                        <span
                          class="visually-hidden"
                        >
                          In bookmarks
                        </span>
                      </button>
                    </div>
                    <div
                      class="place-card__rating rating"
                    >
                      <div
                        class="place-card__stars rating__stars"
                      >
                        <span
                          style="width: 60%;"
                        />
                        <span
                          class="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <h2
                      class="place-card__name"
                    >
                      <a
                        href="/offer/1"
                      >
                        The house among olive 
                      </a>
                    </h2>
                    <p
                      class="place-card__type"
                    >
                      room
                    </p>
                  </div>
                </article>
              </div>
            </li>
          </ul>
        </section>
      </div>
    </main>
    <footer
      class="footer container"
    >
      <a
        class="footer__logo-link"
        href="/"
      >
        <img
          alt="6 cities logo"
          class="footer__logo"
          height="33"
          src="img/logo.svg"
          width="64"
        />
      </a>
    </footer>
  </div>
</div>
`;

exports[`Test routing Render LoginPage when user navigate to '/login' url 1`] = `
<div>
  <div
    class="page page--gray page--login"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="81"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/login"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__login"
                  >
                    Sign in
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--login"
    >
      <div
        class="page__login-container container"
      >
        <section
          class="login"
        >
          <h1
            class="login__title"
          >
            Sign in
          </h1>
          <form
            action="#"
            class="login__form form"
            method="post"
          >
            <div
              class="login__input-wrapper form__input-wrapper"
            >
              <label
                class="visually-hidden"
              >
                E-mail
              </label>
              <input
                class="login__input form__input"
                name="email"
                placeholder="Email"
                type="email"
              />
            </div>
            <div
              class="login__input-wrapper form__input-wrapper"
            >
              <label
                class="visually-hidden"
              >
                Password
              </label>
              <input
                class="login__input form__input"
                name="password"
                placeholder="Password"
                type="password"
              />
            </div>
            <button
              class="login__submit form__submit button"
              type="submit"
            >
              Sign in
            </button>
          </form>
        </section>
        <section
          class="locations locations--login locations--current"
        >
          <div
            class="locations__item"
          >
            <a
              class="locations__item-link"
              href="#"
            >
              <span>
                Amsterdam
              </span>
            </a>
          </div>
        </section>
      </div>
    </main>
  </div>
</div>
`;

exports[`Test routing Render MainPage when user navigate to '/' url 1`] = `
<div>
  <div
    class="page page--gray page--main"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="81"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/login"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__login"
                  >
                    Sign in
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--index "
    >
      <h1
        class="visually-hidden"
      >
        Cities
      </h1>
      <div
        class="tabs"
      >
        <section
          class="locations container"
        >
          <ul
            class="locations__list tabs__list"
          >
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item tabs__item--active"
                href="#"
              >
                <span>
                  Paris
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="#"
              >
                <span>
                  Cologne
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="#"
              >
                <span>
                  Brussels
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="#"
              >
                <span>
                  Amsterdam
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="#"
              >
                <span>
                  Hamburg
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
                href="#"
              >
                <span>
                  Dusseldorf
                </span>
              </a>
            </li>
          </ul>
        </section>
      </div>
      <div
        class="cities"
      >
        <div
          class="cities__places-container container"
        >
          <section
            class="cities__places places"
          >
            <h2
              class="visually-hidden"
            >
              PlaceList
            </h2>
            <b
              class="places__found"
            >
              1 places to stay in Paris
            </b>
            <form
              action="#"
              class="places__sorting"
              method="get"
            >
              <span
                class="places__sorting-caption"
              >
                Sort by
              </span>
              <span
                class="places__sorting-type"
                tabindex="0"
              >
                <svg
                  class="places__sorting-arrow"
                  height="4"
                  width="7"
                >
                  <use
                    xlink:href="#icon-arrow-select"
                  />
                </svg>
              </span>
              <ul
                class="places__options places__options--custom "
              >
                <li
                  class="places__option "
                  tabindex="0"
                >
                  Popular
                </li>
                <li
                  class="places__option "
                  tabindex="0"
                >
                  Price: low to high
                </li>
                <li
                  class="places__option "
                  tabindex="0"
                >
                  Price: high to low
                </li>
                <li
                  class="places__option "
                  tabindex="0"
                >
                  Top rated first
                </li>
              </ul>
            </form>
            <div
              class="cities__places-list places__list tabs__content"
            >
              <article
                class="cities__place-card place-card"
                data-cart-id="1"
              >
                <div
                  class="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  class="cities__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/1"
                  >
                    <img
                      alt="Place image"
                      class="place-card__image"
                      height="200"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
                      width="260"
                    />
                  </a>
                </div>
                <div
                  class=" place-card__info"
                >
                  <div
                    class="place-card__price-wrapper"
                  >
                    <div
                      class="place-card__price"
                    >
                      <b
                        class="place-card__price-value"
                      >
                        €
                        169
                      </b>
                      <span
                        class="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      class="place-card__bookmark-button place-card__bookmark-button--active button"
                      type="button"
                    >
                      <svg
                        class="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlink:href="#icon-bookmark"
                        />
                      </svg>
                      <span
                        class="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    class="place-card__rating rating"
                  >
                    <div
                      class="place-card__stars rating__stars"
                    >
                      <span
                        style="width: 60%;"
                      />
                      <span
                        class="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    class="place-card__name"
                  >
                    <a
                      href="/offer/1"
                    >
                      The house among olive 
                    </a>
                  </h2>
                  <p
                    class="place-card__type"
                  >
                    room
                  </p>
                </div>
              </article>
            </div>
          </section>
          <div
            class="cities__right-section"
          >
            <section
              class="cities__map map"
            >
              <div
                class="leaflet-container leaflet-grab leaflet-touch-drag"
                id="map"
                style="height: 100%; position: relative;"
                tabindex="0"
              >
                <div
                  class="leaflet-pane leaflet-map-pane"
                  style="left: 0px; top: 0px;"
                >
                  <div
                    class="leaflet-pane leaflet-tile-pane"
                  >
                    <div
                      class="leaflet-layer "
                      style="z-index: 1;"
                    >
                      <div
                        class="leaflet-tile-container leaflet-zoom-animated"
                        style="z-index: 18; left: 0px; top: 0px;"
                      >
                        <img
                          alt=""
                          class="leaflet-tile"
                          role="presentation"
                          src="https://b.basemaps.cartocdn.com/rastertiles/voyager/13/4149/2818.png"
                          style="width: 256px; height: 256px; left: -130px; top: -72px;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="leaflet-pane leaflet-shadow-pane"
                  />
                  <div
                    class="leaflet-pane leaflet-overlay-pane"
                  />
                  <div
                    class="leaflet-pane leaflet-marker-pane"
                  >
                    <img
                      alt=""
                      class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                      src="./img/pin.svg"
                      style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: -5px; top: 159px; z-index: 159;"
                      tabindex="0"
                    />
                  </div>
                  <div
                    class="leaflet-pane leaflet-tooltip-pane"
                  />
                  <div
                    class="leaflet-pane leaflet-popup-pane"
                  />
                </div>
                <div
                  class="leaflet-control-container"
                >
                  <div
                    class="leaflet-top leaflet-left"
                  >
                    <div
                      class="leaflet-control-zoom leaflet-bar leaflet-control"
                    >
                      <a
                        aria-label="Zoom in"
                        class="leaflet-control-zoom-in"
                        href="#"
                        role="button"
                        title="Zoom in"
                      >
                        +
                      </a>
                      <a
                        aria-label="Zoom out"
                        class="leaflet-control-zoom-out"
                        href="#"
                        role="button"
                        title="Zoom out"
                      >
                        −
                      </a>
                    </div>
                  </div>
                  <div
                    class="leaflet-top leaflet-right"
                  />
                  <div
                    class="leaflet-bottom leaflet-left"
                  />
                  <div
                    class="leaflet-bottom leaflet-right"
                  >
                    <div
                      class="leaflet-control-attribution leaflet-control"
                    >
                      <a
                        href="https://leafletjs.com"
                        title="A JS library for interactive maps"
                      >
                        Leaflet
                      </a>
                       | © 
                      <a
                        href="https://www.openstreetmap.org/copyright"
                      >
                        OpenStreetMap
                      </a>
                       contributors © 
                      <a
                        href="https://carto.com/attributions"
                      >
                        CARTO
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </section>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
